project-jee8-webapp/src/main/java/fr/pantheonsorbonne/ufr27/miage/jms/conf/JMSProducer.java

```diff
@@ -63,12 +70,12 @@
 	@Produces
 	@Named("bulletin")
 	public Topic getJMSQueue() throws NamingException {
-		return (Topic) JNDI_CONTEXT.lookup("BulletinTopic");
+		return bulletinTopicSupplier.get();
```

Comment:
this is the correct configuration for weld. Note that since we use explicit binding with @Names in the Main class, this is not required in our case.